New rec hit containers

lveldere opened the pull request:
- BaseTrackerRecHit
  BaseTrackerRecHit::initialise qualWord_ in all constructors
- SiTrackerGS*RecHit
  - preparation for arbitrary hit combinations in fast tracking emulation
  - Note: these containers are FastSim-only
- FastSimulation/EgammaElectronAlgos/
  - removed outdated obsolete modules
- TrajectorySeedProducer & TrackCandidateProducer 
  - next step towards arbitrary hit combinations
- SiTrackerGaussianSmearingRecHitConverter
  - adapt to updated recHitContainer
- Miscelaneous
  - several other classes needed to be migrated to the updated recHitConainer

makortel committed:
Update validation campaigns

makortel committed:
Fix typos

makortel committed:
Add validation-style directory structure to trackingCompare.py

makortel committed:
Make canvas a local variable

makortel committed:
Change chi2 y axis to log scale

makortel committed:
Plot new histograms in vertex validation

makortel committed:
Add support for ratio pad

makortel committed:
Draw the uncertainty of denominator, but redraw grid on top of that

makortel committed:
Allow ratio uncertainty to be disable per plot

makortel committed:
Denote ratio denominator uncertainty in legend

makortel committed:
Support rebinning

silviodonato committed:
../python/hltHiggsPostProcessors_cff.py

silviodonato committed:
../python/hltHiggsPostProcessors_cff.py

silviodonato committed:
HLTriggerOffline/Higgs/python/hltHiggsValidator_cfi.py

makortel committed:
Support new-style FastSim pileup DQM file naming

makortel committed:
Allow algoDirMap to be a function in addition to dict

makortel committed:
Monitor again FastSim pileup sample

makortel committed:
Comment out plotting of TrackingParticle for tracks, and non-PU TTbar and gen-vertex for vertices

They were not that useful to be plotted all the time.

makortel committed:
Draw ratio by default in validation scripts

makortel committed:
Add vertexCompare.py script for non-validation vertex plotting

ferencek committed:
degridifying unpacked phi to break degeneracies in angular separations

alja committed:
Changes needed for ROOT6 on linux and osx release.

silviodonato committed:
Update hltHiggsValidator_cfi.py

silviodonato committed:
new path names after b-tag update

ferencek committed:
introduced a pseudo-random sign of the phi shift

venturia committed:
Protect against bunch crossing number = 3564

venturia committed:
Update cfg files to conddb v2

Commit:
Merged refs/pull/9595/head from repository cms-sw

Commit:
first version of cleaned up fastsim rechits

more mess

more mess

Commit:
resolve a last compilation error

Commit:
make hits aware of the hit combination they reside in, and of the multiple simtracks they may originate from

Commit:
new typedefs

Commit:
remove consteness of class SiTrackerGSMatchedRecHit2D members temporary to get things compiled

Commit:
remove outdated analyzer

vkhristenko committed:
Temporary Fix to match Hcal Trig Towers

venturia committed:
New analysis to correlate the BPIX ladder positions with the beamspot position

venturia committed:
Added distributions of the bx and orbit number offsets

venturia committed:
Updated configurations

KiSooLee committed:
HLT name changed

venturia committed:
Added the description of some plugins

venturia committed:
Added the description of some plugins to analyze MC vertices

venturia committed:
Added the description of some configurations

safarzad committed:
add NoNoiseCleand MET trigger

Commit:
get FastSimulation Tracking and TrackingRecHitProducer to compile

Commit:
get FastSimulation/Muon to compile

Commit:
get FastSimulation/EgammaElectronAlgos to compile, remove dead package

Commit:
get SLHCUpgrade package to compile again

safarzad committed:
Add NoNoiseClean Path

safarzad committed:
add New Path to module

Commit:
correct label

safarzad committed:
add PFMET170_v

safarzad committed:
add PFMET170_v

safarzad committed:
add no noise path

safarzad committed:
add no noise path

venturia committed:
Added the description of the plugins to study the multiplicities

venturia committed:
fixed a carriage return

venturia committed:
fixed the instructions to load the library

cfmcginn committed:
Added endoffill to dqm for hi jet, photon, and fulltrack

mdhildreth committed:
add capability of saving Tracking Particles from pileup BX for PreMixing

mdhildreth committed:
Remove digi event content from PREMIX, PREMIXRAW, RAWSIM

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

rmanzoni committed:
fix d0-phi plot under AlcaBeamMonitor/Validation. Now both d0 and phi are computed wrt (0,0,0) and the correlation is preserved

wddgit committed:
Replace getProcessParameterSet function

Replace the function getProcessParameterSet
in DQMServices. This function is being removed
from the CMSSW Framework because it does not work
properly with SubProcesses and causes issues
with multithreading. The replacement should
yield identical behavior to what was originally
there (except now it works in a SubProcess job).

cmsbuild committed:
Merge pull request #9766 from slava77/CMSSW_7_5_X_2015-06-19-1100/niPFrereco

 consistently use momentum at POCA for PFDisplacedVertex momentum computations (same as  #9703  from 74X)

wddgit committed:
Remove function getCentralityBinsFromDB

Remove function getCentralityBinsFromDB
because it uses getProcessParameterSet.
getProcessParameterSet is a Framework function
being removed from CMSSW because it does
not work in SubProcesses and causes issues
for multithreading. Note that
getCentralityBinsFromDB is being removed
instead of being repaired to use an alternative
for the following reasons.

First, nothing in the release calls it. The
only thing that references it is one commented
out line of code.

The function is in a DataFormats package and
accesses the EventSetup. This dependence is
not allowed.

The function is used in modules and pulls
configuration parameters from the top level
parameter set instead of from the parameter
set of the module that uses it. This is only
allowed in very special cases. If this function
was to be repaired in the future it should be
redesigned to get its configuration parameters
from the module in some way.

Also made related fixes to the header includes.

cmsbuild committed:
Merge pull request #9821 from rafaellopesdesa/FixOutOfBoundEnergyUncertaintyPhoton

Fix out of bound array in photon uncertainty calculation

makortel committed:
Add a script to produce validation plots from DQM files given as command line parameters

namapane committed:
Switch to new format for DT uncertainties

cmsbuild committed:
Merge pull request #9798 from makortel/recoTrackSelectorRefactor

Refactor RecoTrackSelector and RecoTrackRefSelector

cmsbuild committed:
Merge pull request #9794 from cms-btv-pog/CommonTMVAEvaluatorForBTagging_from-CMSSW_7_5_X_2015-06-18-1100

Switching b-taggers to a common TMVAEvaluator class

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by @lveldere  for CMSSW_7_6_X.

New rec hit containers

It involves the following packages:

DataFormats/TrackerRecHit2D
FastSimulation/EgammaElectronAlgos
FastSimulation/Muons
FastSimulation/Tracking
FastSimulation/TrackingRecHitProducer
RecoJets/JetAssociationProducers
RecoParticleFlow/PFSimProducer
SimTracker/TrackerHitAssociation


@civanch, @lveldere, @cvuosalo, @ssekmen, @mdhildreth, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@rappoccio, @schoef, @TaiSakuma, @Martin-Grunewald, @threus, @venturia, @mmarionncern, @imarches, @makortel, @acaudron, @lgray, @jdolen, @ferencek, @cerati, @GiacomoSguazzoni, @rovere, @VinInn, @nhanvtran, @wmtford, @dgulhan, @appeltel, @ahinzmann, @gpetruc, @matt-komm, @mariadalfonso, @pvmulder, @bachtis this is something you requested to watch as well.
You can sign-off by replying to this message having '+1' in the first line of your reply.
You can reject by replying  to this message having '-1' in the first line of your reply.
If you are a L2 or a release manager you can ask for tests by saying 'please test' in the first line of a comment.
@degano you are the release manager for this.
You can merge this pull request by typing 'merge' in the first line of your comment.

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

rmanzoni committed:
Added parameters to the migrad functor to supersede EDM defaults (new Minuit changed them after V5.32)

Commit:
remove outdated fastsim pileup option

Commit:
remove the genmix customisation function

Commit:
remove the pileupproducer package

Commit:
remove gen-mixing option from FamosProducer

Commit:
remove forgotten include file

Commit:
remove vertex smearing generators used in gen-mixing

davidlange6 committed:
Merge pull request #9744 from Dr15Jones/makeSimHitPrinterThreadSafe

Made file access in SimHitPrinter thread safe

davidlange6 committed:
Merge pull request #9745 from Dr15Jones/makeEcalSrFlagStaticThreadSafe

Made static EcalSrFlag::srfNames fully const

cmsbuild committed:
Merge pull request #9809 from ferencek/DegridifiedUnpackedPhi_from-CMSSW_7_5_X_2015-06-10-1100

Degridifying unpacked phi to break degeneracies in angular separations for packed candidates

cmsbuild commented:
-1 
Tested at: 36f9e5a12275d005746339c8de1312abf1caf857
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-06-24-1100/src/FastSimulation/Tracking/plugins/SimTrackIdProducer.cc 
Leaving library rule at CommonTools/RecoAlgos
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-06-24-1100/src/FastSimulation/Tracking/plugins/TrackCandidateProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-06-24-1100/src/FastSimulation/Tracking/plugins/TrajectorySeedProducer.cc 
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-06-24-1100/src/SUSYBSMAnalysis/HSCP/plugins/HSCPValidator.cc:49:0:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2015-06-24-1100/poison/FastSimulation/Tracking/test/FastTrackAnalyzer.h:1:2: error: #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
 #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
  ^
@@@@ ----> OK  EDM Class Version 
>> Building edm plugin tmp/slc6_amd64_gcc491/src/FastSimulation/Tracking/test/testGeneralTracks/libtestGeneralTracks.so
>> Building edm plugin tmp/slc6_amd64_gcc491/src/FastSimulation/Tracking/test/testTrackingIterations/libtestTrackingIterations.so
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/48/summary.html

wmtan committed:
Capabilities libraries are obsolete

alja committed:
Ignore LD_LIBRARY_PATH from the parent process.

alja committed:
Update m_source variable in checkFieldInfo(). The bug manifested in track-propagator field.

Commit:
remove obsolete includes of no-longer-existing header file

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #9855 was updated. @civanch, @lveldere, @cvuosalo, @ssekmen, @mdhildreth, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cschomak committed:
Include d<angle> vs. d<translation> plots in geometry comparison plotter

davidlange6 committed:
Merge pull request #9859 from lveldere/fastsim-removeGenMix-cmsDriver

FastSim: remove outdated pileup option from cmsDriver

cmsbuild committed:
Merge pull request #9865 from matteosan1/hltdqm_75X

Update to handle trigger paths with DiSC prefix

cmsbuild committed:
Merge pull request #9860 from lveldere/fastsim-removeGenMix-famosproducer

Fastsim: remove gen mixing from famosproducer

cmsbuild committed:
Merge pull request #9834 from MilanoBicocca-pix/CMSSW_7_6_X_fix_beamspot_d0_phi_dqm

Offline DQM: fix d0-phi plot under AlcaBeamMonitor/Validation

cmsbuild committed:
Merge pull request #9817 from silviodonato/DQMWHbb

DQM for HLT_Ele27_WPLoose_Gsf_WHbbBoost_v1 - 75X

cmsbuild committed:
Merge pull request #9805 from silviodonato/DQM-higgs-btag-fix

Fix of path names in HLT Higgs validation code - 75X

cmsbuild committed:
Merge pull request #9804 from silviodonato/DQMbtagfix

Fix path name in HLT BTV valdiation code

makortel committed:
Allow dEdx plots be ignored for FastSim (don't exist anymore in 750pre4 for FS)

cmsbuild committed:
Merge pull request #9753 from cfmcginn/HI50nsDQMUpdate

Added endoffill to dqm for hi jet, photon, and fulltrack

cmsbuild committed:
Merge pull request #9749 from barvic/cscdqm_run2_histos_patch_75X

CSC DQM Run2 histograms adjustments and fixes for CMSSW_7_5_X

cmsbuild commented:
+1
Tested at: 86efdd3a9409fe6a2af88565832d7673e2dc7016
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/5695/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/5695/summary.html

lveldere commented:
+1

civanch commented:
+1

cmsbuild committed:
Merge pull request #9872 from wmtan/NoCapabilities

Capabilities libraries are obsolete

cmsbuild committed:
Merge pull request #9774 from safarzad/myPath

add no noise clean path to SUSYBSM_inclusiveMET_cff.py  & SUSYBSM_postProcessor_cff.py & SusyExoValidation_cff.py

danduggan committed:
online DQM updates in 76x: EvD configs and small fixes

cmsbuild committed:
Merge pull request #9883 from namapane/DTSwitchToNewRecoUncert_76X

Switch to new format for DT uncertainties

cmsbuild committed:
Merge pull request #9795 from vkhristenko/tpfix_75

Temporary Fix to match Hcal Trig Towers

Commit:
GEMCSCSegAlgoRR: initialize xe and ye

    DataFormats/GeometryVector/interface/extBasic3DVector.h:78:14: error:
    'ye' may be used uninitialized in this function
    [-Werror=maybe-uninitialized]
         v{x,y,z,w}{}
              ^
    RecoLocalMuon/GEMCSCSegment/plugins/GEMCSCSegAlgoRR.cc:217:14: note:
    'ye' was declared here
        float xe, ye, ze = 0.0;

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

slava77 commented:
Lukas, please document expected changes in physics performance.

There are changes in  the fastsim workflows tested in jenkins

e.g. ZEE fastsim 135.4 workflow
![all_oldvsnew_zeefs13wf135p4c_log10recotracks_generaltracks__hlt_obj_pt](https://cloud.githubusercontent.com/assets/4676718/8361183/2d58657c-1b37-11e5-8c54-e48baee3feb6.png)
it looks like there are less tracks 
![all_oldvsnew_zeefs13wf135p4c_recotracks_generaltracks__hlt_obj_algo](https://cloud.githubusercontent.com/assets/4676718/8361266/b9e804fc-1b37-11e5-97ff-d918f4e908a9.png)

... or equivalently less PF charged hadrons
![all_oldvsnew_zeefs13wf135p4c_recopfcandidates_particleflow__hlt_obj_particleid](https://cloud.githubusercontent.com/assets/4676718/8361236/81765efc-1b37-11e5-93c5-b33f5d5d158c.png)

slava77 commented on DataFormats/TrackerRecHit2D/src/SiTrackerGSMatchedRecHit2D.cc:
```diff
@@ -1,58 +1,9 @@
-   const SiTrackerGSMatchedRecHit2D* otherCast = static_cast<const SiTrackerGSMatchedRecHit2D*>(other);
+					      SharedInputType what) const
+{
+  const SiTrackerGSMatchedRecHit2D * otherCasted = dynamic_cast<const SiTrackerGSMatchedRecHit2D*>(other);
```
would static_cast still work? (probably faster)

lveldere replied:
Static cast probably won't always work because there are 2 fastsim hit types.
We are going to work on the inheritance structure of the fastsim rechits,
which will avoid the dynamic casts, also in other places.

slava77 commented on DataFormats/TrackerRecHit2D/src/classes_def.xml:
```diff
@@ -5,15 +5,19 @@
-  <class name="SiTrackerGSRecHit2D" ClassVersion="11">
+  <class name="SiTrackerGSRecHit2D" ClassVersion="13">
+   <version ClassVersion="13" checksum="1916407081"/>
+   <version ClassVersion="12" checksum="1369340574"/>
```
we should not skip versions due to some intermediate change
- remove version 12 and change 13 to 12

lveldere replied:
fixed

lveldere commented:
Thanks Slava

It's not really expected...
However I still would like to have it merged,
since several new developments depend on this pr.

I think this is justified because we are early in 76,
and since it's only FastSim that moves.
We'll start pinning down the issue soon.

Cheers

Lukas

alja committed:
Update filed info in FF.

slava77 commented on DataFormats/TrackerRecHit2D/src/classes_def.xml:
```diff
@@ -5,15 +5,19 @@
+  <class name="SiTrackerGSMatchedRecHit2D" ClassVersion="15">
+    <version ClassVersion="15" checksum="933733669"/>
+    <version ClassVersion="14" checksum="3149196391"/>
+    <version ClassVersion="13" checksum="1553184391"/>
```
similarly, v14 is fake and is not needed; remove 14 and then change 15 to 14

lveldere replied:
fixed

alja committed:
Reset field estimate on new data file.

slava77 commented on FastSimulation/TrackingRecHitProducer/plugins/SiTrackerGaussianSmearingRecHitConverter.cc:
```diff
@@ -620,41 +594,30 @@ void SiTrackerGaussianSmearingRecHitConverter::produce(edm::Event& e, const edm:
+    for(size_t hindex = 0;hindex < recHitCombinations->at(cindex).size();++hindex){
+      std::cout << recHitCombinations->at(cindex)[hindex].hitCombinationId() << " " << recHitCombinations->at(cindex)[hindex].id() << " " << recHitCombinations->at(cindex)[hindex].simTrackId(0) << std::endl;
+    }
+  }
```
is this commented out printout still needed ?

lveldere replied:
fixed

slava77 commented on FastSimulation/TrackingRecHitProducer/plugins/SiTrackerGaussianSmearingRecHitConverter.cc:
```diff
@@ -1399,14 +1297,17 @@ SiTrackerGaussianSmearingRecHitConverter::matchHits(
+	std::cout << rit->det()->index() << std::endl;
+	std::cout << rit->simTrackId(0);
+	exit(0);
+	*/
```
another bunch of commented out stuff

lveldere replied:
fixed

davidlange6 committed:
Merge pull request #9896 from davidlt/fix-xe-and-ye-GEMCSCSegAlgoRR

GEMCSCSegAlgoRR: initialize xe and ye

cmsbuild committed:
Merge pull request #9891 from danduggan/EvDinput_76x

online DQM updates in 76x: EvD configs and small fixes

cmsbuild committed:
Merge pull request #9882 from makortel/trackingValidation_v2

Update tracking MC validation scripts

cmsbuild committed:
Merge pull request #9875 from alja/75-field

75x Fireworks: port bugfix in mag field determination from 74

cmsbuild committed:
Merge pull request #9870 from wddgit/removegetCentralityBinsFromDB

Remove function getCentralityBinsFromDB

cmsbuild committed:
Merge pull request #9869 from wddgit/getProcessParameterSetDQMServices

Replace getProcessParameterSet function

davidlange6 committed:
Merge pull request #9772 from Dr15Jones/uninitValueJetDeltaRValueMapProducer-7_5

Fix an uninitialized value in JetDeltaRValueMapProducer

cmsbuild committed:
Merge pull request #9858 from MilanoBicocca-pix/CMSSW_7_6_X_Minuit_EDM_Fix

BeamSpot Fitter: added parameters to the migrad functor to supersede EDM defaults

cmsbuild committed:
Merge pull request #9776 from ngrenz/CMSSW_7_5_X_ngrenz_string_makeMaps

reduce creation of string objects and use char* instead of string

cmsbuild committed:
Merge pull request #9741 from cms-l1t-offline/l1t-FED-vector-HF-uppgrade-bugfix-75x

L1t fed vector hf uppgrade bugfix 75x

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

mmusich committed:
two liner to make AlignmentTrackSelectorModule multi-threaded friendly

Commit:
avoid unnecessary data format version number

Commit:
remove unnecessary printouts

mmusich committed:
additional fixes

davidlange6 committed:
Merge pull request #9888 from davidlange6/mem_reduct_744p1

fixing memory leak in L1 CSCTrackFinder

davidlange6 committed:
Merge pull request #9773 from venturia/dpganalysis_sistriptools-75x

Fix for orbit number = 3564, update of the documentation and other improvements

cmsbuild committed:
Merge pull request #9793 from KiSooLee/HLT_DQM_75X

HLT name changed

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #9855 was updated. @civanch, @lveldere, @cvuosalo, @ssekmen, @mdhildreth, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

deguio committed:
minor fixes online applications. same as #9623

alja committed:
Fix module name for Jet collection.

alja committed:
Change Jet collection module.

cmsbuild commented:
+1
Tested at: 1fb1812db221112c777548e2099f7622eb61a00d
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/54/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/54/summary.html

cmsbuild committed:
Merge pull request #9927 from dmitrijus/dqm_not_init_75x

Dqm not init 75x

cmsbuild committed:
Merge pull request #9910 from mmusich/fixAlignmentTrackSelectorMultiThreadedFriendly76X

Fix alignment track selector multi threaded friendly for 76X (fwdport of #9906)

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

davidlange6 committed:
Merge pull request #9737 from mdhildreth/TPUpdatesForPreMixing

TrackingParticle updates for pre mixing

cmsbuild committed:
Merge pull request #9930 from alja/75x-config

75x Fireworks: port Fireworks configuration changes from 74

cmsbuild committed:
Merge pull request #9759 from mmarionncern/METPAT_METCor75X

New MET correction/uncertainty tool

slava77 commented:
Looking at 5.1 workflow (ttbar run 1 setup)

the MultiTrack validation is completely broken now
![wf5p1_mtv](https://cloud.githubusercontent.com/assets/4676718/8391653/9b5df146-1c8f-11e5-8895-317fe6ecd0a9.png)

There is a pretty large reduction in the higher pt tracks.
My earlier ZEE plot was mostly the Z->ee tracks at higher pt. This one should be mainly hadrons
![all_test9855-histatvsorig-histat_ttbarfswf5p1c_log10recotracks_generaltracks__hlt_obj_pt](https://cloud.githubusercontent.com/assets/4676718/8391473/91cb2824-1c8a-11e5-8c42-19adb262084a.png)


changes at the particle flow level (after track quality selections or calorimeter matching) confirm that most of the change is in the charged hadrons

![all_test9855-histatvsorig-histat_ttbarfswf5p1c_log10recopfcandidates_particleflowtmp__hlt_obj_pt80](https://cloud.githubusercontent.com/assets/4676718/8391524/f21cdb40-1c8b-11e5-8e45-e60307f215a1.png)

... 
looking at btag validation plots, it seems to me even the baseline is pretty bad already (TCHE for b jets should go much higher)
![wf5p1_tche_b_udsg_eff](https://cloud.githubusercontent.com/assets/4676718/8391661/c147eec0-1c8f-11e5-82fb-a2a66a70f70d.png)

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/9855	1fb1812
- regular fullsim reco is not affected
- fastsim is pretty badly broken (see comments above)
- signing off at request from Lukas to not hold this PR

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

davidlange6 committed:
remove forgotten usage of fastsim pileup

davidlange6 committed:
Merge pull request #9942 from davidlange6/removeFSPU

remove forgotten usage of fastsim pileup

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

lveldere commented:
Thanks  a lot Slava
I try to look into it today.

Commit:
Merged refs/pull/9855/head from repository cms-sw

cmsbuild committed:
Merge pull request #9923 from deguio/fixOnline75x

minor fixes online applications. same as #9623

cmsbuild committed:
Merge pull request #9879 from cschomak/CMSSW_7_6_X_GeometryComparison

Include d<angle> vs. d<translation> plots in geometry comparison plotter

cmsbuild committed:
Merge CMSSW_7_5_X into CMSSW_7_6_X.

Commit:
add script to conveniently run harvesting of track validation plots

lveldere committed:
Update harvestTrackValidationPlots.py

lveldere committed:
use subprocess.call() w/o shell=True, fix preamble

lveldere committed:
clarified error message

Commit:
move variables and function common to SiTrackerGSMatched and SiTrackerGSRecHit to base class, make SiTrackerGS* objects self-aware, correct initialisation of SiTrackerGSRecHits, fix sharesInput function

lveldere commented:
Thanks again Slava for checking this so carefully.

I found the issue:
The SiTrackerGSRecHit::sharesInput function returned "true" to often,
which is fixed now.

Along with that, I moved some variables and functions from SiTrackerGSRecHit2D and SiTrackerGSMatchedRecHit2D to the base class (also FastSim specific), 
which allowed to get rid of the dynamic casts.

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #9855 was updated. @civanch, @lveldere, @cvuosalo, @ssekmen, @mdhildreth, @monttj, @cmsbuild, @slava77, @vadler can you please check and sign again.

cmsbuild commented:
+1
Tested at: 3753b4af931150924ea65515e032ffc653145a4b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/187579/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9855/187579/summary.html

Commit:
Merged refs/pull/9968/head from repository cms-sw

lveldere commented:
Closing for a while

Something is still broken about the truth matching for FastSim.

@slava77: any idea what the big red piece of pie for DQM is?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_7_6_X_2015-06-30-2300+9855/5854/25.0_TTbar+TTbar+DIGI+RECOAlCaEle+HARVEST+ALCATT/

slava77 commented:
On 7/1/15 9:28 AM, lveldere wrote:
> Closing for a while
> 
> Something is still broken about the truth matching for FastSim.
> 
> @slava77 <https://github.com/slava77>: any idea what the big red piece
> of pie for DQM is?
> https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_7_6_X_2015-06-30-2300+9855/5854/25.0_TTbar+TTbar+DIGI+RECOAlCaEle+HARVEST+ALCATT/
> 

TimerService (you can click on the pies to navigate deeper)
this is expected to differ since times between runs are not really that
stable


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/9855#issuecomment-117694522>.
>

Commit:
fix fastsim rechit event id, adapt TrackerHitAssociator to edited fastsim rechit format

lveldere commented:
truth matching is fixed,
I performed the track validation sequence on 10 fastsim events before and after changes and obtain identical results

lveldere commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #9855 was updated. @smuzaffar, @civanch, @diguida, @lveldere, @cvuosalo, @ssekmen, @mdhildreth, @monttj, @cmsbuild, @alja, @franzoni, @Dr15Jones, @cerminar, @deguio, @slava77, @danduggan, @vadler, @mmusich, @davidlange6 can you please check and sign again.

lveldere commented:
superseded by #10018

slava77 commented:
-1
... Lukas, you can close this

lveldere commented:
sorry Slava, I pushed the wrong button... :-)
